src/v/cloud_topics/frontend/frontend.cc

```diff
@@ -642,8 +642,47 @@ frontend::validate_fetch_offset(
   bool reading_from_follower,
   model::timeout_clock::time_point deadline) {
     if (reading_from_follower && !_partition->is_leader()) {
-        // TODO: implement follower fetching for cloud topics
-        co_return std::unexpected(frontend_errc::not_leader_for_partition);
+        std::optional<frontend_errc> ec = std::nullopt;
+        auto lso = last_stable_offset();
+        if (!lso.has_value()) {
+            ec = frontend_errc::offset_not_available;
+        }
+
+        model::offset leader_hwm;
+        kafka::offset available_to_read;
+
+        if (!ec.has_value()) {
+            leader_hwm
+              = _partition->get_offset_translator_state()->from_log_offset(
+                _partition->leader_high_watermark());
+            available_to_read = std::min(
+              model::offset_cast(leader_hwm), lso.value());
```

Comment:
Why do we need to do this? replicated_partition does not, and the fetch handler should already clamp to LSO for read committed fetches. Instead available to read is std::min(leader_high_watermark(), log_end_offset());

Reply:
I agree, we should use `min(leader_hw, local_end_offset)`

Reply:
fixed